In running C2D README, got data_service_provider.py "result retrieval unsuccessful"

**Describe the bug**
I was going through the C2D README, on the [PR](https://github.com/oceanprotocol/ocean.py/pull/913/files#diff-c8e25d9af3e6334fa1b6fcdb49e959cc8f5456fb4b2aad9c71879ed98650637f) to make things remote. Most changes were just about pointing to remote network, vs local. I was able to get all to run, until just before the end, at step 7 where "7. Bob monitors logs / algorithm output"

It went through the first part -  a loop until results are done - and found results so got `successful = True`.

Then when doing the next step:
```python
output = ocean.compute.compute_job_result_logs(
    DATASET_asset, compute_service, job_id, bob_wallet
)[0]
```

... it failed. 

Here's the traceback:
```text
Traceback (most recent call last):
  File "/home/trentmc/code/ocean.py/ocean_lib/ocean/ocean_compute.py", line 134, in compute_job_result_logs
    result = self._data_provider.compute_job_result_logs(
  File "/home/trentmc/code/ocean.py/venv/lib/python3.10/site-packages/enforce_typing/decorator.py", line 29, in wrapper
    return func(*args, **kwargs)
  File "/home/trentmc/code/ocean.py/ocean_lib/data_provider/data_service_provider.py", line 419, in compute_job_result_logs
    assert result, "result retrieval unsuccessful"
AssertionError: result retrieval unsuccessful
```

Of the four results coming back from `compute_job_result_logs()`, the first three gave a non-empty `result` (such that "assert result" passes). The fourth has a `result` of `b""`. That's what caused the failure.

--- a/ocean_lib/data_provider/data_service_provider.py
+++ b/ocean_lib/data_provider/data_service_provider.py
@@ -416,7 +416,9 @@ class DataServiceProvider(DataServiceProviderBase):
             result = DataServiceProvider.compute_job_result(
                 job_id, i, dataset_compute_service, consumer
             )
-            assert result, "result retrieval unsuccessful"
+
+            if result_type != "publishLog":
+                assert result, f"result retrieval unsuccessful. i={i}"
 
             # Extract algorithm output
             if result_type == log_type:

